cdk/l3-cdk-resources/atlas-basic/src/index.ts

```diff
@@ -1,98 +1,424 @@
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly username?: string;
+}
+/**
+ * @description
+ * @export
+ * @interface IpAccessListProps
+ */
+export interface IpAccessListProps {
+  /**
+     * @description
+     * @type {ipAccessList.AccessListDefinition[]}
+     * @memberof IpAccessListProps
+     */
+  readonly accessList: ipAccessList.AccessListDefinition[];
+  /**
+     * @description
+     * @type {string}
+     * @memberof IpAccessListProps
+     */
+  readonly projectId?: string;
+  /**
+     * @description
+     * @type {number}
+     * @memberof IpAccessListProps
+     */
+  readonly totalCount?: number;
+  /**
+     * @description
+     * @type {ipAccessList.ListOptions}
+     * @memberof IpAccessListProps
+     */
+  readonly listOptions?: ipAccessList.ListOptions;
+}
+/**
+ * @description
+ * @export
+ * @interface ApiKeyDefinition
+ */
+export interface ApiKeyDefinition {
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly privateKey?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly publicKey?: string;
 
+}
+/**
+ * @description
+ * @export
+ * @interface AtlasBasicProps
+ * @extends {cdk.StackProps}
+ */
+export interface AtlasBasicProps extends cdk.StackProps{
+  /**
+     * @description
+     * @type {ApiKeyDefinition}
+     * @memberof AtlasBasicProps
+     */
+  readonly apiKeys : ApiKeyDefinition;
+  /**
+     * @description
+     * @type {ProjectProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly projectProps: ProjectProps;
+  /**
+     * @description
+     * @type {ClusterProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly clusterProps: ClusterProps;
+  /**
+     * @description
+     * @type {DatabaseUserProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly dbUserProps?: DatabaseUserProps;
+  /**
+     * @description
+     * @type {IpAccessListProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly ipAccessListProps?: IpAccessListProps;
+}
+/**
+ * @description
+ * @export
+ * @class AtlasBasic
+ * @extends {cdk.Stack}
+ */
 export class AtlasBasic extends cdk.Stack {
+  /**
+   * Creates an instance of AtlasBasic.
+   * @param {Construct} scope
+   * @param {string} id
+   * @param {AtlasBasicProps} props
+   * @memberof AtlasBasic
+   */
   constructor(scope: Construct, id: string, props: AtlasBasicProps) {
     super(scope, id, props);
-    // Create a new MongoDB Atlas Project
-    // const mProject = new project.CfnProject(this, 'project-'.concat(id), {
-    // //    name: defaults.projectName,
-    //     apiKeys: props.apiKeys,
-    //     ...props.projectProps
-    // });
+    //Create a new MongoDB Atlas Project
+  const mProject = new project.CfnProject(this, 'project-'.concat(id), {
+      apiKeys: props.apiKeys,
+      name: props.projectProps.name || projectDefaults.projectName,
+      ...props.projectProps
+  });
```

Comment:
if the customer runs two times without providing the clusterName this will fail since a cluster with the name `projectDefaults.projectName` would be already there. Could we change the default project name to have a random part such as cfn-cluster-1233 maybe? tagging @Zuhairahmed to decide the default cluster name to use.

Reply:
suggest we keep simple: "atlas-cluster-[random number]" and "atlas-project-[random number]" if no other concerns